public/index.php

```diff
@@ -43,6 +43,20 @@
  * and fires up an environment-specific bootstrapping.
  */
 
+/**
+ * A real path with identical slashes in Unix/Windows
+ */
+function _realpath(string $path): bool|string
```

Comment:
I would put this inside your `normalize_path` function.

Reply:
Maybe. But they are created in different areas. The problem now is duplicates - they are needed in the earliest call. `normalize_path()` is available later

Reply:
Oh, these are slightly different functions. `normalize_path()` can take a nonexistent or relative path. Why add `realpath()` to it?